l10n_it_fatturapa_out_rc/wizard/wizard_export_e_invoice.py

```diff
@@ -6,11 +6,38 @@
     AnagraficaType,
     IndirizzoType
 )
+from odoo.fields import first
 
 
 class WizardExportFatturapa(models.TransientModel):
     _inherit = "wizard.export.fatturapa"
 
+    def group_invoices_by_partner(self):
+        """
+        Group Reverse charge self-invoices one by one.
+
+        We are sending invoices to ourselves,
+        and XMLs containing self-invoices are currently managed as
+        if there was one self-invoice in each XML
+        (see `fatturapa.attachment.in.is_self_invoice`
+        and `fatturapa.attachment.in.linked_invoice_id_xml`)
+        """
+        invoice_ids = self.env.context.get('active_ids', False)
+        invoices = self._get_rc_invoices(invoice_ids)
+        invoices_with_rc, _other_invoices = self._split_rc_invoices(invoices)
+        if invoices == invoices_with_rc:
+            # These are self-invoices, they are all for ourselves
+            partner = first(invoices_with_rc).partner_id
+            invoices_by_partner = {
+                partner: [
```

Comment:
Ma così non fai un unico XML per tutte le fatture?

Reply:
> Ma così non fai un unico XML per tutte le fatture?

No perché `invoices_by_partner[partner]` è una lista di liste; in particolare, ogni lista è `invoice_with_rc.ids`, quindi la lista degli `id` di `invoice_with_rc` (in pratica è equivalente a `[invoice_with_rc.id]`).

Verrà poi generato un XML per ogni lista contenuta in `invoices_by_partner[partner]`, quindi un XML per ogni fattura.